### PR TITLE
User management

### PR DESCRIPTION
**Pull request by Thomas-Julien-SAHUT**
-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Services/AuthenticationService.cs.ejs [hunk 1]
```diff
@@ -34,7 +34,7 @@ namespace <%= namespace %>.Client.Xamarin.Core.Services
 
         public bool IsAuthenticated { get; set; }
         public UserModel CurrentUser { get; set; }
-        private JwtToken JwtToken { get; set; }
+        public JwtToken JwtToken { get; set; }
```
Comment (Zenrac) [inline generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Services/AuthenticationService.cs.ejs:37]: Shouldn't be public

File: generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Services/IAuthenticationService.cs.ejs [hunk 2]
```diff
@@ -23,6 +23,7 @@ namespace <%= namespace %>.Client.Xamarin.Core.Services
     {
         bool IsAuthenticated { get; set; }
         UserModel CurrentUser { get; set; }
+        JwtToken JwtToken { get;}
```
Comment (Zenrac) [inline generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Services/IAuthenticationService.cs.ejs:26]: Don't add this to interface

File: generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Services/UserEntityService.cs.ejs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+namespace <%= namespace %>.Client.Xamarin.Core.Services
+{
+    public class UserEntityService : IUserEntityService<UserModel>
+    {
+        private const string AuthorizationHeader = "Authorization";
+    
+        protected readonly HttpClient _httpClient;
+        private IAuthenticationService _authenticationService;
+    
+        protected JwtToken JwtToken { get; set; }
```
Comment (Zenrac) [inline generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Services/UserEntityService.cs.ejs:32]: Same

File: generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Services/UserEntityService.cs.ejs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        protected string BaseUrl = "api/users";
+        public UserEntityService(HttpClient httpClient, IAuthenticationService authenticationService)
+        {
+            _httpClient = httpClient;
+            _authenticationService = authenticationService;
+
+            _httpClient.BaseAddress = new Uri(Configuration.BaseUri);
+            JwtToken = authenticationService.JwtToken;
+    
+            if (JwtToken != null)
```
Comment (Zenrac) [inline generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Services/UserEntityService.cs.ejs:42]: Same

File: generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Services/UserEntityService.cs.ejs [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        private IAuthenticationService _authenticationService;
+    
+        protected JwtToken JwtToken { get; set; }
+        protected string BaseUrl = "api/users";
+        public UserEntityService(HttpClient httpClient, IAuthenticationService authenticationService)
+        {
+            _httpClient = httpClient;
+            _authenticationService = authenticationService;
+
+            _httpClient.BaseAddress = new Uri(Configuration.BaseUri);
```
Comment (Zenrac) [inline generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Services/UserEntityService.cs.ejs:39]: Same

File: generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/ViewModels/MenuViewModel.cs.ejs [hunk 6]
```diff
@@ -25,12 +28,17 @@ namespace <%= namespace %>.Client.Xamarin.Core.ViewModels
         private readonly IMvxNavigationService _navigationService;
         private readonly IAuthenticationService _authenticationService;
 
+        public IMvxCommand ShowMyUsersCommand => new MvxAsyncCommand(ShowMyUsersCommandClicked);
+        public IMvxCommand ShowMyEntitiesCommand => new MvxAsyncCommand(ShowMyEntitiesCommandClicked);
```
Comment (Zenrac) [inline generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/ViewModels/MenuViewModel.cs.ejs:32]: Remove this

File: generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/ViewModels/MenuViewModel.cs.ejs [hunk 7]
```diff
@@ -40,6 +48,16 @@ namespace <%= namespace %>.Client.Xamarin.Core.ViewModels
 
         // jhipster-needle-add-entity-command - JHipster will add commands
 
+        private async Task ShowMyEntitiesCommandClicked()
```
Comment (Zenrac) [inline generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/ViewModels/MenuViewModel.cs.ejs:51]: Remove this

File: generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/ViewModels/MenuViewModel.cs.ejs [hunk 8]
```diff
@@ -40,6 +48,16 @@ namespace <%= namespace %>.Client.Xamarin.Core.ViewModels
...
+        {
+            await _navigationService.Navigate<MyEntityViewModel>();
```
Comment (Zenrac) [inline generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/ViewModels/MenuViewModel.cs.ejs:53]: MyEntityViewModel doesnt exist anymore

File: generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Views/MenuPage.xaml.ejs [hunk 9]
```diff
@@ -43,6 +43,18 @@
                         <TapGestureRecognizer Tapped="ToggleClicked" Command="{Binding ShowWelcomeCommand}"/>
                     </Grid.GestureRecognizers>
                 </Grid>
+                <Grid HorizontalOptions="FillAndExpand" VerticalOptions="StartAndExpand" Padding="8" BackgroundColor="LightBlue" IsVisible="{Binding IsAdmin}">
+                    <Label Text="Test User" />
+                    <Grid.GestureRecognizers>
+                        <TapGestureRecognizer Tapped="ToggleClicked" Command="{Binding ShowMyUsersCommand}" CommandParameter="TODO"/>
+                    </Grid.GestureRecognizers>
+                </Grid>
+                <Grid HorizontalOptions="FillAndExpand" VerticalOptions="StartAndExpand" Padding="8" BackgroundColor="LightBlue" IsVisible="{Binding IsConnected}">
```
Comment (Zenrac) [inline generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Views/MenuPage.xaml.ejs:52]: Remove this

File: generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Views/MenuPage.xaml.ejs [hunk 10]
```diff
@@ -43,6 +43,18 @@
                         <TapGestureRecognizer Tapped="ToggleClicked" Command="{Binding ShowWelcomeCommand}"/>
                     </Grid.GestureRecognizers>
                 </Grid>
+                <Grid HorizontalOptions="FillAndExpand" VerticalOptions="StartAndExpand" Padding="8" BackgroundColor="LightBlue" IsVisible="{Binding IsAdmin}">
+                    <Label Text="Test User" />
+                    <Grid.GestureRecognizers>
+                        <TapGestureRecognizer Tapped="ToggleClicked" Command="{Binding ShowMyUsersCommand}" CommandParameter="TODO"/>
```
Comment (Zenrac) [inline generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Views/MenuPage.xaml.ejs:49]: Remove CommandParameter

File: generators/client/templates/xamarin/test/Project.Client.Test/Pages/TestPages/MyEntityViewModelTest.cs.ejs [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+using FluentAssertions;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using AutoFixture;
+
+namespace <%= namespace %>.Client.Test.Pages.TestPages
+
+{
+    [TestClass]
+    public class MyEntityViewModelTest : MvxIoCSupportingTest
```
Comment (Zenrac) [inline generators/client/templates/xamarin/test/Project.Client.Test/Pages/TestPages/MyEntityViewModelTest.cs.ejs:32]: MyEntity doesnt exist anymore

**Review by Zenrac — CHANGES_REQUESTED** [review]
MyEntity doesnt exist anymore. You dont handled conflicts correctly

File: generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/ViewModels/MenuViewModel.cs.ejs (hunk 7 above)
Comment (Thomas-Julien-SAHUT) [reply generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/ViewModels/MenuViewModel.cs.ejs:51]: Why ?

File: generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Views/MenuPage.xaml.ejs (hunk 10 above)
Comment (Thomas-Julien-SAHUT) [reply generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Views/MenuPage.xaml.ejs:49]: Why ?

Comment (Zenrac) [reply generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Views/MenuPage.xaml.ejs:49]: Never used

File: generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/ViewModels/MenuViewModel.cs.ejs (hunk 7 above)
Comment (Zenrac) [reply generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/ViewModels/MenuViewModel.cs.ejs]: MyEntity doesnt exist anymore

File: generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/App.cs.ejs [hunk 12]
```diff
@@ -52,6 +52,11 @@ namespace <%= namespace %>.Client.Xamarin.Core
             Mvx.IoCProvider.RegisterSingleton<IAuthenticationService>(authenticationService);
             var registerService = new RegisterService(httpClient, log);
             Mvx.IoCProvider.RegisterSingleton<IRegisterService>(registerService);
+            var myEntityService = new AbstractEntityService<AbstractEntityModel>(httpClient, "api/myentities");
```
Comment (Zenrac) [inline generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/App.cs.ejs:55]: Remove this

File: generators/client/files-xamarin.js [hunk 13]
```diff
@@ -392,6 +464,44 @@ const files = {
...
+                    file: 'Project.Client.Test/Pages/TestPages/LoginTest.cs',
+                    renameTo: generator => `${generator.clientTestProject}/LoginTest.cs`,
+                },
+            ],
+        },
+        {
+            path: CLIENT_TEST_DIR,
+            templates: [
+                {
+                    file: 'Project.Client.Test/Pages/TestPages/MyEntityViewModelTest.cs',
```
Comment (Zenrac) [inline generators/client/files-xamarin.js:490]: Doesnt exist anymore

File: generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Project.Client.Xamarin.Core.csproj.ejs [hunk 14]
```diff
@@ -33,6 +33,12 @@
     <Compile Update="Views\RegisterView.xaml.cs">
       <DependentUpon>RegisterView.xaml</DependentUpon>
     </Compile>
+    <Compile Update="Views\MyEntityView.xaml.cs">
+      <DependentUpon>MyEntityView.xaml</DependentUpon>
```
Comment (Zenrac) [inline generators/client/templates/xamarin/src/Project.Client.Xamarin.Core/Project.Client.Xamarin.Core.csproj.ejs:37]: Doesnt exist

File: generators/client/templates/xamarin/test/Project.Client.Test/Pages/TestPages/MyEntityViewModelTest.cs.ejs [hunk 15]
```diff
@@ -0,0 +1,197 @@
...
+using FluentAssertions;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using AutoFixture;
+
+namespace <%= namespace %>.Client.Test.Pages.TestPages
+
+{
+    [TestClass]
+    public class MyAbstractEntityViewModelTest : MvxIoCSupportingTest
```
Comment (Zenrac) [inline generators/client/templates/xamarin/test/Project.Client.Test/Pages/TestPages/MyEntityViewModelTest.cs.ejs:32]: MyEntityView doesnt exist too

**Comment by nicolas63** [comment]
@Zenrac any updates ?

**Pull request closed without merging**